File: src/test/java/net/bbelovic/adventofcode/day1/Day1PuzzlePart1Test.java
package net.bbelovic.adventofcode.day1;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day1PuzzlePart1Test {
    @ParameterizedTest
    @MethodSource("testData")
    void
    should_compute_floor_from_input(String input, long expectedResult) {
        var day1 = new Day1PuzzlePart1();
        long actualResult = day1.solve(input);
        assertEquals(expectedResult, actualResult);
    }

    static Stream<Arguments> testData() {
        return Stream.of(
                Arguments.of("(((", 3L),
                Arguments.of("(())", 0L),
                Arguments.of("))(((((", 3L),
                Arguments.of("()()", 0L),
                Arguments.of("())", -1L),
                Arguments.of(")))", -3L),
                Arguments.of(")())())", -3L)
        );
    }
}

File: aoc-2020/src/test/kotlin/net/bbelovic/adventofcode/year2020/day1/Day1PuzzlePart1Test.kt
package net.bbelovic.adventofcode.year2020.day1

import net.bbelovic.adventofcode.InputReader
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

const val PART1_EXPECTED_RESULT = 982464L

class Day1PuzzlePart1Test {

    @Test
    fun `should compute product of two entries that make 2020 - example data`() {
        val inputEntries = listOf("1721", "979", "366", "299", "675", "1456")
        val expectedResult = 514579L
        val actualResult = Day1PuzzlePart1Recursion(2).solve(inputEntries)
        assertThat(actualResult).isEqualTo(expectedResult)
    }

    @Test
    fun `should compute product of two entries that make 2020`() {
        val puzzle = Day1PuzzlePart1()
        val input = InputReader().readAllLines("input1.txt")
        val actualResult = puzzle.solve(input)
        assertThat(actualResult).isEqualTo(PART1_EXPECTED_RESULT)
    }
}